pdp-server/src/api/authzen/evaluations.rs

```diff
@@ -0,0 +1,376 @@
+        (None, Some(r)) => Some(r),
+        _ => None,
+    };
+
+    // Get action from evaluation or default
+    let action = match (&evaluation.action, &defaults.action) {
+        (Some(a), _) => Some(a),
+        (None, Some(a)) => Some(a),
+        _ => None,
+    };
+
+    // Merge context from evaluation and default
+    let mut context = defaults.context.clone().unwrap_or_default();
+    if let Some(eval_context) = &evaluation.context {
+        context.extend(eval_context.clone());
+    }
+
+    // Return None if any required field is missing
+    if subject.is_none() || resource.is_none() || action.is_none() {
+        return None;
+    }
+
+    // Create a complete AccessEvaluationRequest
+    let req = AccessEvaluationRequest {
+        subject: subject.unwrap().clone(),
+        resource: resource.unwrap().clone(),
+        action: action.unwrap().clone(),
+        context: Some(context),
+    };
+
+    // Convert to AllowedQuery
+    Some(AllowedQuery::from(req))
+}
+
+// Convert AllowedResult to EvaluationResult
+fn convert_to_evaluation_result(allowed_result: AllowedResult) -> EvaluationResult {
+    let mut context: Option<HashMap<String, serde_json::Value>> = None;
+
+    if let Some(debug) = allowed_result.debug {
+        context = Some(debug);
+    } else if let Some(query) = allowed_result.query {
+        context = Some(query);
+    }
+
+    EvaluationResult {
+        decision: allowed_result.allow,
+        context,
+    }
+}
+
+#[utoipa::path(
+    post,
+    path = "/access/v1/evaluations",
+    tag = AUTHZEN_TAG,
+    request_body = AccessEvaluationsRequest,
+    params(
+        ("Authorization" = String, Header, description = "Authorization header"),
+        ("X-Request-ID" = String, Header, description = "Request Identifier"),
+    ),
+    responses(
+        (status = 200, description = "Access evaluations completed successfully", body = AccessEvaluationsResponse),
+        (status = 400, description = "Bad Request"),
+        (status = 401, description = "Unauthorized"),
+        (status = 403, description = "Forbidden"),
+        (status = 500, description = "Internal server error")
+    )
+)]
+pub async fn access_evaluations_handler(
+    State(state): State<AppState>,
+    Json(request): Json<AccessEvaluationsRequest>,
+) -> Response {
+    // Handle the case with no evaluations (backward compatibility)
+    if request.evaluations.is_empty() {
+        return ApiError::bad_request("No evaluations provided").into_response();
+    }
+
+    // Convert each evaluation to an AllowedQuery
+    let queries: Vec<AllowedQuery> = request
+        .evaluations
+        .iter()
+        .filter_map(|eval| convert_to_allowed_query(eval, &request))
+        .collect();
+
+    // If any conversion failed, return an error
+    if queries.len() != request.evaluations.len() {
+        return ApiError::bad_request("One or more evaluations is missing required fields")
+            .into_response();
+    }
```

Comment:
why "abstract" the error of conversion ? it makes the user harder to understand which fields are missing

Reply:
Added detailed info about each evaluation and it's missing fields